x/execution/client/rest/query.go

```diff
@@ -58,7 +67,28 @@ func queryListHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
 			return
 		}
 
+		execs := make([]*execution.Execution, 0)
+		if err := cliCtx.Codec.UnmarshalJSON(res, &execs); err != nil {
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		sortFunc := sort.Interface(execution.ByBlockHeight(execs))
+		for _, tag := range tags {
+			if tag == "sort='reverse'" {
+				sortFunc = sort.Reverse(sortFunc)
+			}
+		}
+		sort.Sort(sortFunc)
+
+		start, end := client.Paginate(len(execs), page, limit, limit)
+		if start < 0 || end < 0 {
+			execs = make([]*execution.Execution, 0)
```

Comment:
We should return error if the pagination boundaries are invalid (so in that case check if page/limit <= 0), otherwise returning empty list.

`s/execs = make([]*execution.Execution, 0)/execs = nil/`

Reply:
the page and limit <= 0 error is managed by `rest.ParseHTTPArgs(r)`:
```go
} else if page <= 0 {
	return tags, page, limit, errors.New("page must greater than 0")
}
```
```go
} else if limit <= 0 {
	return tags, page, limit, errors.New("limit must greater than 0")
}
```
---
the problem with returning `execs = nil` is it's not a list anymore in the json response:
```json
{"height":"757","result":null}
```
so we will have issue with clients.
i would keep it as it is:
```
{"height":"773","result":[]}
```

i replaced the `make([]*execution.Execution, 0)` by simply `[]*execution.Execution{}`